Detectors/PHOS/calib/src/BadChannelMap.cxx

```diff
@@ -0,0 +1,65 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "PHOSBase/Geometry.h"
+#include "PHOSCalib/BadChannelMap.h"
+
+#include "FairLogger.h"
+
+#include <TH2.h>
+
+#include <iostream>
+
+using namespace o2::phos;
+
+TH2* BadChannelMap::getHistogramRepresentation(int module) const
```

Comment:
This is a good example where it would be better returning a unique ptr, which is the suggested practice. Otherwise this could leak memory. Alternatively you could pass the TH2 to be filled as an argument?